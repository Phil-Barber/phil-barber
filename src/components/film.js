import React from 'react';
import Layout from './layout';

const Film = ({ children, filmTitle, filmPoster }) => (
  <Layout>
    <h1>{filmTitle}</h1>
    {children}
    <img src={filmPoster} alt={`${filmTitle} Poster`} />
  </Layout>
);

export default Film;
