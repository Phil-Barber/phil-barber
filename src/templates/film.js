import React from 'react';
import { graphql } from 'gatsby';
import { Review } from './review';

const FilmTemplate = ({ data }) => {
  const post = data.markdownRemark;
  const featuredImgFluid = post.frontmatter.featuredImage.childImageSharp.fluid;
  const { released, dateCompleted, director, starring } = post.frontmatter;
  const details = [
    {
      attr: 'Released',
      value: released,
    },
    {
      attr: 'Watched',
      value: dateCompleted,
    },
    {
      attr: 'Director',
      value: director.join(', '),
    },
    {
      attr: 'Starring',
      value: starring.join(', '),
    },
  ];

  return (
    <Review
      imageFluid={featuredImgFluid}
      details={details}
      title={post.frontmatter.title}
      description={post.frontmatter.excerpt}
    >
      <div dangerouslySetInnerHTML={{ __html: post.html }} />
    </Review>
  );
};

export default FilmTemplate;

export const query = graphql`
  query FilmQuery($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      html
      frontmatter {
        title
        released
        dateCompleted
        director
        starring
        featuredImage {
          childImageSharp {
            fluid(maxWidth: 300, maxHeight: 400, quality: 100) {
              ...GatsbyImageSharpFluid
            }
          }
        }
      }
    }
  }
`;
