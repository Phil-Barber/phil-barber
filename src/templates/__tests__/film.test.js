import React from 'react';
import { shallow } from 'enzyme';
import { FilmTemplate } from '../film';

describe('FilmTemplate', () => {
  it('Matches snapshot', () => {
    const container = shallow(
      <FilmTemplate
        data={{
          markdownRemark: {
            html: '<div>This is the film review</div>',
            frontmatter: {
              title: 'FilmTemplate title',
              released: '2019',
              dateCompleted: '2018-02-11',
              director: ['Director 1', 'Director 2'],
              starring: ['Some Gal', 'Some Guy'],
              featuredImage: {
                childImageSharp: {
                  fluid: 'ImageFluid',
                },
              },
            },
          },
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
