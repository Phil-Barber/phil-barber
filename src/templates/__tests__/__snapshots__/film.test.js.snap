// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmTemplate Matches snapshot 1`] = `
<Review
  details={
    Array [
      Object {
        "attr": "Released",
        "value": "2019",
      },
      Object {
        "attr": "Watched",
        "value": "2018-02-11",
      },
      Object {
        "attr": "Director",
        "value": "Director 1, Director 2",
      },
      Object {
        "attr": "Starring",
        "value": "Some Gal, Some Guy",
      },
    ]
  }
  imageFluid="ImageFluid"
  title="FilmTemplate title"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>This is the film review</div>",
      }
    }
  />
</Review>
`;
